### PR TITLE
fix(image-v2,dynamic-media): deliver animated gifs from dynamic media

### LINKED ISSUE #1307: Animated Gifs via Dynamic Media in Image Component
## Bug Report
Animated Gifs are not supported in the Image Component when delivering images via Dynamic Media

**Current Behavior**
After enabling Dynamic Media via Settings and Policies, i can upload an animated gif to the assets folder. Adding it to the Component and publishing it to Dynamic Media works as expected. However, the image, when displayed on the page, is not animated anymore but the still image (first frame) of the animated gif is shown.

**Expected behavior/code**
When adding an animated gif, i expect the animation to be shown on the page it is added to.

**Environment**
- AEM 6.4.8.1
- Core Components version 2.13.0
- JRE version jdk1.8.0_261-amd64/jre

**Possible Solution**
When the image is served, it uses the
/is/image
server path to embed the image in the Component (Check ImageImpl.java line 179)
As discussed, among others, in https://experienceleaguecommunities.adobe.com/t5/adobe-experience-manager-assets/handle-gifs-on-scene-7-mode/qaq-p/358593 animated gifs have to be delivered via /is/content

### PR DESCRIPTION
**Pull request by jkoebner**
deliver animated gifs from dynamic media content servlet

deliver animated gifs from scene7 as content instead of as image since the image servlet path does not support compression and delivery of animated gifs and only returns the first still image of the animated gif

1307

| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #1307 
| Patch: Bug Fix?          | Yes
| Minor: New Feature?      | No
| Major: Breaking Change?  | No
| Tests Added + Pass?      | No tests added
| Documentation Provided   | No
| Any Dependency Changes?  | No
| License                  | Apache License, Version 2.0

Added changes to image servlet path assembly both in ImageImpl (v2) and DAMAssetPostProcessor.
in both files, checked the asset metadata to retrieve information if the image is an animated gif. If so, the servlet path that is added into the image component's content and that is later used to display the image is adapted.
For all regular images, the image servlet path should be used (/is/image). Animated Gifs can only be rendered when retrieved from the content servlet (/is/content)

### CODE REVIEW

**Comment by jkoebner** [comment]
I am in contact with stakeholders from DM to see if the image servlet is going to support delivery of animated gifs. If delivery of animated gifs via /is/image is to be added, the gifs should be (for semantics reasons and possible functional reasons) delivered from that servlet.

**Comment by oufimtse** [comment]
Hi @jkoebner, thanks a lot for your interest in the product and the actual contribution.
We have reviewed the code changes and did a quick test in QA environment, and the change looks good both from code perspective and functionality-wise.

The only thing which does not allow to merge this PR "as is" is automation. 
Would you be able to extend the following two unit tests?
https://github.com/jkoebner/aem-core-wcm-components/blob/issue/1307/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImplTest.java
https://github.com/jkoebner/aem-core-wcm-components/blob/issue/1307/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/servlets/DMAssetPostProcessorTest.java

**Comment by jkoebner** [comment]
looking into it. thanks!

**Comment by jkoebner** [comment]
@oufimtse The Unit Tests have been extended and should be altight. not sure why the codecov report states 1.26% decrease of code coverage. can you maybe restart the report generation or provide support on what i need to adapt?

**Comment by oufimtse** [comment]
Hi @jkoebner, may I ask you to update your forked branch to the latest changes from the development branch [1] please? The core components team has stopped using CircleCI and switched to GitHub actions meanwhile.
I hope that this update will give us representative PR validation results here, and we'll be able to finalize the discussion about the tests.
Thanks.

[1] https://github.com/adobe/aem-core-wcm-components/commit/ecf3409deb1ee88d27a390d2ed4636e203cadd7c#diff-87ee5504a3e25ac558b343724c905f2f7949e8cec3d92b9c4300bb922afa164f

**Comment by jkoebner** [comment]
@oufimtse done. thanks for the int. all should be fine now!

**Review by oufimtse — APPROVED** [review]

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java [hunk 1]
```diff
@@ -175,8 +176,12 @@ protected void initModel() {
                         boolean isWCMDisabled =  (com.day.cq.wcm.api.WCMMode.fromRequest(request) == com.day.cq.wcm.api.WCMMode.DISABLED);
                         String dmServerUrl;
                         if (!isWCMDisabled) {
-                            //for Author
-                            dmServerUrl = "/is/image/";
+                            if (asset.getMetadataValue(Scene7Constants.PN_S7_TYPE).equals(Scene7AssetType.ANIMATED_GIF.getValue())) {
```
Comment (vladbailescu) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java:179]: Please import the constants from `DMAssetPostProcessor.java`!

**Review by vladbailescu — CHANGES_REQUESTED** [review]

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java (hunk 1 above)
Comment (jkoebner) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java:179]: hello there. i am trying to add your suggested change, but DMAssetPostProcessor only has two (private!) constants: IMAGE_SERVER_PATH and CONTENT_SERVER_PATH
i could make them public and use them here. is that what you are referring to?

Comment (vladbailescu) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java:179]: Yes, please reuse them!

Comment (jkoebner) [reply bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/ImageImpl.java:181]: OK - should be done now. shall i resolve this conversation?

**Review by vladbailescu — APPROVED** [review]

**Comment by vladbailescu** [comment]
@jkoebner LGTM but you will need to fix the merge conflicts before I can apply the changes.

**Comment by jkoebner** [comment]
@vladbailescu it all should be fine now. hope i added all the required changes

**Pull request merged**